src/models/archaeologist.ts

```diff
@@ -73,7 +72,12 @@ export class Archaeologist {
   async initNode(arg: { config: PublicEnvConfig; web3Interface: Web3Interface }) {
     if (this.listenAddressesConfig) {
       const { signalServerList } = this.listenAddressesConfig!;
-      this.listenAddresses = genListenAddresses(signalServerList, this.peerId.toJSON().id);
+      this.listenAddresses = genListenAddresses(
+        signalServerList,
+        this.peerId.toJSON().id,
+        false,
+        process.env.DOMAIN
```

Comment:
Noted in the TODO, but calling out here --

Followup task will add validation for this env var, and potentially store the domain on the arch model.